engine/runtime/src/main/scala/org/enso/compiler/context/SuggestionDiff.scala

```diff
@@ -0,0 +1,130 @@
+package org.enso.compiler.context
+import org.enso.polyglot.Suggestion
+import org.enso.polyglot.data.{These, Tree}
+import org.enso.polyglot.runtime.Runtime.Api
+
+object SuggestionDiff {
+
+  /** Compute difference between the two suggestion trees.
+    *
+    * @param prev the tree before
+    * @param current the tree after
+    * @return the tree with updates
+    */
+  def compute(
+    prev: Tree.Root[Suggestion],
+    current: Tree.Root[Suggestion]
+  ): Tree.Root[Api.SuggestionUpdate] =
+    Tree
+      .zipBy(prev, current)(compare)
+      .map(diff)
+
+  /** Compare two suggestions for equality. */
+  private def compare(a: Suggestion, b: Suggestion): Boolean =
+    a.name == b.name &&
+    a.module == b.module &&
+    Suggestion.Kind(a) == Suggestion.Kind(b) &&
+    Suggestion.SelfType(a) == Suggestion.SelfType(b)
```

Comment:
We need to find matching suggestions when joining the previous suggestion tree with the current. Since there is no stable identifier, we assume there are no two suggestions in the same module in the same scope with the same name of the same kind.
I updated the doc.